src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/FixedActiveThreadsExecutor.kt

```diff
@@ -138,14 +129,14 @@ internal class FixedActiveThreadsExecutor(private val nThreads: Int, runnerHash:
         }
         // Park with timeout until the result is set or the timeout is passed.
         val currentThread = Thread.currentThread()
-        if (results[iThread].compareAndSet(null, Thread.currentThread())) {
+        if (results[iThread].compareAndSet(null, currentThread)) {
```

Comment:
Please improve the documentation/comments in this file.

Reply:
I extended the doc-string for `submitAndAwait` method (after the change it has new more complicated contracts). 
All other public methods in the `FixedActiveThreadsExecutor` class are already documented. Besides that, their semantics haven't changed. 

What else do you want me to improve in this file?